pylearn2/models/mlp.py

```diff
@@ -2987,15 +2995,27 @@ def initialize_transformer(self, rng):
         """
         if self.irange is not None:
             assert self.sparse_init is None
-            self.transformer = conv2d.make_random_conv2D(
-                irange=self.irange,
-                input_space=self.input_space,
-                output_space=self.detector_space,
-                kernel_shape=self.kernel_shape,
-                subsample=self.kernel_stride,
-                border_mode=self.border_mode,
-                rng=rng)
-        elif self.sparse_init is not None:
+
+            if cudnn2d:
+                self.transformer = cudnn2d.make_random_conv2D(
+                    irange=self.irange,
+                    input_space=self.input_space,
+                    output_space=self.detector_space,
+                    kernel_shape=self.kernel_shape,
+                    subsample=self.kernel_stride,
+                    border_mode=self.border_mode,
+                    pad=self.pad,
+                    rng=rng)
+            else:
+                self.transformer = conv2d.make_random_conv2D(
+                    irange=self.irange,
+                    input_space=self.input_space,
+                    output_space=self.detector_space,
+                    kernel_shape=self.kernel_shape,
+                    subsample=self.kernel_stride,
+                    border_mode=self.border_mode,
+                    rng=rng)
```

Comment:
Here, we completely ignore `pad`, we should have a meaningful error message instead if `pad` is specified but `cudnn2d` is not available.